src/controllers/user.js

```diff
@@ -12,7 +15,62 @@ const getProfile = async (req, res) => {
         .json(currentUser);
     }
   } catch (error) {
-    return res.sendStatus(500).json({ message: error.message });
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+const getUserProducts = async (req, res) => {
+  try {
+    let { postType, SearchTitles } = req.query;
+    const filteringQueries = { $or: [] };
+    const finalQuery = { $and: [] };
+
+    if (postType) {
+      if (typeof postType !== 'object') postType = postType.split();
+
+      postType.forEach((type) => {
+        if (type === 'Donated') {
```

Comment:
instead of the reptetive if condition, objects can be used in order to change the value of postTypeDepending on the type so for example you can have an object like this 
```
POST_TYPE_SELECTOR = {
Donated: {postType:'Donate'},
Requested:{postType:'Request', isTransactionCompleted:false}
}
```
and then use this selector and spread the value of it inside the filteringQueries
like this:
```
      filteringQueries.$or.push({
            publisher: req.params.userId,
            {...POST_TYPE_SELECTOR[type]}
           });
```
now its a little bit clearner and whenever you have a new option u just add it to the SELECTOR object and its handeled automatically without needing to add another if condition

@okurezgi  @khatibAmjad  Hopefully this makes it a bit easier and cleaner